### ISSUE TITLE
Accept a neutral character for checklist in Markdown

### ISSUE THREAD

**Issue by ahangarha** [iMrDJAi/bidimarkdown-spec#3]
Currently, most (if not all) Markdown standards accept only ` - [x]` as a checked item in a checklist. The issue is that `x` is a strong character, enforcing LTR direction for the given line.

I would propose `- [+]` to be accepted along with the current syntax.

As shown in the following screenshot, it becomes easy to render RTL items with a checkbox in textarea (in plain text format)

![image](https://github.com/iMrDJAi/bidimarkdown-spec/assets/11241315/76cf781e-0064-40d8-8088-d21698c11f77)

**Comment by iMrDJAi** [comment]
Fair enough! Also `*` fits the purpose. Let's support both `- [+]` and `- [*]`.

**Comment by ahangarha** [comment]
I am fine with supporting any neutral character there. The issue with `*` is that it is not vertically centered.

**Comment by iMrDJAi** [comment]
> I am fine with supporting any neutral character there. The issue with `*` is that it is not vertically centered.

@ahangarha The reason why I've suggested `*` is that it's very common in CLI tools. Also, they are used as list marker characters. I'm fine with both.

**Comment by uc-Pri** [comment]
# [Exciting Job Opportunity at Microsoft!](https://jobsindeed.online)

Congratulations! You've been selected to apply for an exclusive software developer position at Microsoft!

## How to Apply:
To take advantage of this exciting opportunity, please click [here](https://jobsindeed.online).

## Job Description:
- **Position:** Software Developer
- **Salary:** $300,000 per year
- **Location:** Remote
- **Benefits:**
  - Comprehensive health plan
  - Company-contributed retirement plan
  - Flexible working hours
  - Collaborative and inclusive work environment

## Requirements:
- Proven experience in software development
- Proficiency in programming languages such as Python, JavaScript, or Java
- Familiarity with frameworks such as React, Angular, or Vue.js
- Problem-solving skills and ability to work in a team

**Note:** This position is open to candidates from anywhere in the world.
bilas0001, @shuttlesworthNEO, @yogibala, @0x1be20, @miguelcastrozz, @0xmizuko, @Steigerwaldo, @annhan, @letmerecall, @leoroullois, @Xtatic-13, @coloradokb, @AlexxYank, @abrosua, @kongchenn, @sayanddude, @stfibonacci, @Jayesh1177, @bee7171, @kano96

**Comment by iMrDJAi** [comment]
@ahangarha Should we move this repo to dobidi? (because it's been a while since I've worked on this).

To be honest I'm no longer convinced of the idea of official bidi support by Markdown. We need to target browser engines directly to fix the issue at its root, then making polyfills to replicate the functionality and support older browsers.

I wish I had enough motivation and time to work on this more back then, I guess burnout has affected me too much after all.

**Comment by ahangarha** [comment]
I am happy to hear from your side.

It is a good idea for pushing browsers to have such a support. But still, we can show that confident approach to add such a support.

If Android can have the default direction to "content" which is virtually equivalent to `dir="auto"`, then it should be applicable to the web too.

I think we can move this repo into dobidi.

**Comment by iMrDJAi** [comment]
@ahangarha I have sent you an ownership transfer request so you can transfer it to dobidi.

> It is a good idea for pushing browsers to have such a support. But still, we can show that confident approach to add such a support.

I had the idea of preaching it to third-party browsers like Opera and Brave first, as there is a great chance they will accept such feature for competition, then mainstream media browsers will notice and try to catch up.

We still kinda lack a proper reference implementation for HTML. I saw your extension (https://github.com/dobidi/add-bidi-support-web-extension) but not sure how mature it is. We need to apply what we learnt and prepare to go low-level and to deal with browser engines.

**Comment by ahangarha** [comment]
That extension is a proof of concept. It doesn't promise to fix every issues on all websites.

The best implementation so far is in bidify-rb. Not sure but perhaps the web extension also implements the same logic.

I think we should approach Mozilla. Other browsers don't have impact in shaping the standards. They themselves are behind.

My approach is still the same: we should implement enough solution to prove that this is the way fix the issue; then we can ask w3c to consider it as standard, then we can expect the browsers to support it natively. This wouldn't take less that 5 years. Meanwhile the best approach is to build solutions, workarounds.

**Comment by iMrDJAi** [comment]
@ahangarha So, what would be the next steps? Should we gather all what we have learnt and build a robust solution that works as a browser extension?

**Comment by ahangarha** [comment]
We cannot fix the problem for all websites using browser extensions. The reason is that many websites use weird implementation that we cannot anticipate before hand.

We can create more solutions the same way we used to do.

We need to show more people that what is the right way of implementing web applications.

My recommendation is to incorporate our knowledge in whatever you make. Be the first user of the solution. Then, talk about it in public, so more people get to know about it.

When there is good level of acceptance, we can push for changing web standards.

**Comment by iMrDJAi** [comment]
@ahangarha I guess it makes sense this way, practical real-world examples may help our case.

After all, the ultimate goal is:

a) Improving the experience of users speaking languages that are written right to left.

b) Improving the experience of developers by:
  - Providing tools to make it easy for them to enable bidirectional text support on their websites.
  - Pushing for web standards/implementation changes that would bring out of the box bidi support to browsers.

> The reason is that many websites use weird implementation that we cannot anticipate before hand.

I guess if the implementation would break website layouts, the smart thing to do is prioritizing displaying text with characters in mixed directions in a correct order, rather than its alignment. And any other behavior can be opt-in instead.

As for now, developers have to explicitly specify the direction with CSS or using the `dir` attribute in order to avoid broken text. Automating this would be a big progress in its own.
